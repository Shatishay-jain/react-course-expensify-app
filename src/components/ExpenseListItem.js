import React from 'react';
import {connect} from 'react-redux';
import {Link} from 'react-router-dom';
import moment from 'moment';
import numeral from 'numeral';
const ExpenseListItem =({description,amount,createdAt,id})=>(
    <div>
        
        <Link to={`./edit/${id}`} >
        <h3>{description}</h3>
        </Link>
        <p>
        Rs-
    {
        ( amount / 100)
    
    
    
    }
        ,
        {moment(createdAt).format('MMMM Do YYYY')}
        </p>
        
        
    </div>
);

export default ExpenseListItem;
