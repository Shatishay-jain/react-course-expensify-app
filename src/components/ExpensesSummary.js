import React from 'react';
import {connect} from 'react-redux';

import selectExpenses from '../selectors/expenses';
import selectExpensesTotal from '../selectors/expenses-total';

export const ExpressSummary=({expenseCount,expensesTotal})=>{
    const ExpenseWord = expenseCount===1?'expense':'expenses';
    const formattedExpenseTotal=expensesTotal/100;
    return(
        <div>
            <h1>
             viewing {expenseCount} {ExpenseWord} totalling Rs-{formattedExpenseTotal}   
            </h1>
        </div>
    );

};
const mapStateToProps=(state)=>{
    const visibleExpenses =selectExpenses(state.expense,state.filters);
    return{
        expenseCount:visibleExpenses.length,
        expensesTotal:selectExpensesTotal(visibleExpenses)
    };


};
export default connect(mapStateToProps)(ExpressSummary);