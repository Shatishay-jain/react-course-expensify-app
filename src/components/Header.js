import React from 'react';
import { BrowserRouter , Route , Switch , Link , NavLink} from 'react-router-dom';
const Header=()=>(
    <header>
        <h1>
            Expensify
        </h1>
            <NavLink to='/' activeClassName ='is-active' exact={true}>Go home </NavLink>    
            <NavLink to='/create' activeClassName ='is-active' exact={true}>AddexpensePage </NavLink>
            
            <NavLink to='/help' activeClassName ='is-active' exact={true}>HelpPage 1</NavLink>

    </header>
);
export default Header;