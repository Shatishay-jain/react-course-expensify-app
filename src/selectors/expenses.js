import moment from 'moment';

const getVisibleExpenses=(expense,{text,sortBy,startDate,endDate})=>{
    return expense.filter((expenses)=>{
        const createdAtMoment=moment(expenses.createdAt);
        const startDateMatch = startDate?startDate.isSameOrBefore(createdAtMoment, 'day') :true;
        const endDateMatch = endDate? endDate.isSameOrAfter(createdAtMoment, 'day'):true;
        
        const textMatch = expenses.description.toLowerCase().includes(text.toLowerCase());
      //  console.log(text.length,text);
        //if(text.length==0)
        //textMatch=true;
        return startDateMatch && endDateMatch && textMatch;
    
        }).sort((a,b)=>{
            if(sortBy==='date')
            {
                return a.createdAt<b.createdAt?1:-1
            }
            else if(sortBy==='amount')
            {
                return a.amount<b.amount?1:-1
            }
    });
    }
export default getVisibleExpenses;