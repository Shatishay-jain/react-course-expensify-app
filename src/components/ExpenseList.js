import React from 'react';
import {connect} from 'react-redux';
import ExpenseListItem from './ExpenseListItem';
import selectExpense from '../selectors/expenses';

export const ExpenseList=(props)=>(
    <div>
       {
           props.expense.length===0?(<p>No expenses</p>):(
            props.expense.map((expenses)=>{
                return <ExpenseListItem key={expenses.id} {...expenses}/>;
            })
           )
       }
        
       
    </div>
);
const mapStateToProps=(state)=>{
    return{
        expense : selectExpense(state.expense,state.filters)
    }
};
export default  connect(mapStateToProps)(ExpenseList);

