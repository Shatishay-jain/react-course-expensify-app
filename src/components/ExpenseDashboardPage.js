import React from 'react';
import ExpenseList from './ExpenseList';
import ExpenseListFilters from './ExpenseListFilters';
import ExpensesTotal from './ExpensesSummary';
const ExpenseDashboardPage=()=>(

    <div>
        <ExpensesTotal/>
        <ExpenseListFilters/>
        <ExpenseList/>
    </div>
);
export default ExpenseDashboardPage;