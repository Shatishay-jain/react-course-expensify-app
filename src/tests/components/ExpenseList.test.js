import React from 'react';
import {shallow} from 'enzyme';
import {ExpenseList} from '../../components/ExpenseList';
import expenses from '../fixtures/expenses';

test('should render expenseList from expense',()=>{
    const wrapper=shallow(<ExpenseList expense={expenses} />);
    expect(wrapper).toMatchSnapshot();
});
test('should render nothing for empty expense',()=>{
    const wrapper=shallow(<ExpenseList expense={[]} />);
    expect(wrapper).toMatchSnapshot();
});