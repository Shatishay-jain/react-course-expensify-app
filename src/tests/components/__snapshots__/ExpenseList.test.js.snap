// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render expenseList from expense 1`] = `
<div>
  <ExpenseListItem
    amount={195}
    createdAt={0}
    description="gum"
    id="1"
    key="1"
    note=""
  />
  <ExpenseListItem
    amount={1095}
    createdAt={-345600000}
    description="Rent"
    id="2"
    key="2"
    note=""
  />
  <ExpenseListItem
    amount={1905}
    createdAt={345600000}
    description="coffe"
    id="3"
    key="3"
    note=""
  />
</div>
`;

exports[`should render nothing for empty expense 1`] = `
<div>
  <p>
    No expenses
  </p>
</div>
`;
